components/script/dom/htmlmediaelement.rs

```diff
@@ -1650,7 +1818,138 @@ impl HTMLMediaElementContext {
+            FetchMetadata::Filtered { unsafe_, .. } => unsafe_,
+        });
+
+        // Step 14.5 of https://html.spec.whatwg.org/multipage/#img-environment-changes
+        if let Some(metadata) = metadata.as_ref() {
+            if let Some(ref content_type) = metadata.content_type {
+                let mime: Mime = content_type.clone().into_inner().into();
+                if mime.type_() == mime::MULTIPART && mime.subtype().as_str() == "x-mixed-replace" {
+                    self.aborted = true;
+                    let elem = self.elem.root();
+                    elem.poster_frame_canceller.borrow_mut().cancel();
+                    elem.poster_generation_id
+                        .set(elem.poster_generation_id.get() + 1);
+                }
+            }
+        }
+
+        let status_code = metadata
+            .as_ref()
+            .and_then(|m| m.status.as_ref().map(|&(code, _)| code))
+            .unwrap_or(0);
+
+        self.status = match status_code {
+            0 => Err(NetworkError::Internal(
+                "No http status code received".to_owned(),
+            )),
+            200...299 => Ok(()),
+            _ => Err(NetworkError::Internal(format!(
+                "HTTP error code {}",
+                status_code
+            ))),
+        };
+    }
+
+    fn process_response_chunk(&mut self, payload: Vec<u8>) {
+        if self.aborted {
+            // An error was received previously, skip processing the payload.
+            return;
+        }
+
+        if self.status.is_ok() {
+            self.image_cache
+                .notify_pending_response(self.id, FetchResponseMsg::ProcessResponseChunk(payload));
+        }
+    }
+
+    fn process_response_eof(&mut self, response: Result<ResourceFetchTiming, NetworkError>) {
+        self.image_cache
+            .notify_pending_response(self.id, FetchResponseMsg::ProcessResponseEOF(response));
+        let elem = self.elem.root();
+        elem.delay_load_event(false);
+    }
+
+    fn resource_timing_mut(&mut self) -> &mut ResourceFetchTiming {
+        &mut self.resource_timing
+    }
+
+    fn resource_timing(&self) -> &ResourceFetchTiming {
+        &self.resource_timing
+    }
+
+    fn submit_resource_timing(&mut self) {
+        network_listener::submit_timing(self)
+    }
+}
+
+impl ResourceTimingListener for PosterFrameFetchContext {
+    fn resource_timing_information(&self) -> (InitiatorType, ServoUrl) {
+        let initiator_type = InitiatorType::LocalName(
+            self.elem
+                .root()
+                .upcast::<Element>()
+                .local_name()
+                .to_string(),
+        );
+        (initiator_type, self.url.clone())
+    }
+
+    fn resource_timing_global(&self) -> DomRoot<GlobalScope> {
+        (document_from_node(&*self.elem.root()).global())
+    }
+}
+
+impl PreInvoke for PosterFrameFetchContext {
+    fn should_invoke(&self) -> bool {
+        !self.aborted
```

Comment:
We have logic in the methods for what to do in error cases, so we should always invoke the methods.